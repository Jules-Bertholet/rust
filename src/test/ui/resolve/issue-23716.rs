static foo: i32 = 0;

fn bar(foo: i32) {}
//~^ ERROR referencing statics in patterns is experimental

mod submod {
    pub static answer: i32 = 42;
}

use self::submod::answer;

fn question(answer: i32) {}
//~^ ERROR referencing statics in patterns is experimental
fn main() {
}
