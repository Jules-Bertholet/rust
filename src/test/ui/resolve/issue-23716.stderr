error[E0658]: referencing statics in patterns is experimental
  --> $DIR/issue-23716.rs:3:8
   |
LL | fn bar(foo: i32) {}
   |        ^^^
   |
   = help: add `#![feature(static_in_pattern)]` to the crate attributes to enable

error[E0658]: referencing statics in patterns is experimental
  --> $DIR/issue-23716.rs:12:13
   |
LL | fn question(answer: i32) {}
   |             ^^^^^^
   |
   = help: add `#![feature(static_in_pattern)]` to the crate attributes to enable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
